packages/edit-navigation/src/components/name-editor/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useRef, useState, useContext } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { TextControl } from '@wordpress/components';
+import {
+	IsMenuEditorFocused,
+	useMenuEntity,
+	useNavigationEditorMenu,
+} from '../../hooks';
+import { useInstanceId } from '@wordpress/compose';
+
+export function NameEditor() {
+	const [ isMenuNameEditFocused, setIsMenuNameEditFocused ] = useContext(
+		IsMenuEditorFocused
+	);
+
+	const { menuName, menuId } = useNavigationEditorMenu();
+	const { editMenuName } = useMenuEntity( menuId );
```

Comment:
`useMenuEntity` could get the `menuId` from context, which would make it easier to combine both `useNavigationEditorMenu` and `useMenuEntity`.